app/ui/legacy/src/main/java/com/fsck/k9/ui/settings/general/GeneralSettingsDataStore.kt

```diff
@@ -131,8 +131,8 @@ class GeneralSettingsDataStore(
         when (key) {
             "language" -> appLanguageManager.setAppLanguage(value)
             "theme" -> setTheme(value)
-            "message_compose_theme" -> K9.messageComposeTheme = stringToSubTheme(value)
-            "messageViewTheme" -> K9.messageViewTheme = stringToSubTheme(value)
+            "message_compose_theme" -> generalSettingsManager.setMessageComposeTheme(stringToSubTheme(value))
+            "messageViewTheme" -> generalSettingsManager.setMessageViewTheme(stringToSubTheme(value))
```

Comment:
This will save the settings twice, once via `GeneralSettingsManager` and once via `GeneralSettingsDataStore.saveSettings()`. I suggest extracting this code to separate methods and setting a boolean to skip the save operation in `GeneralSettingsDataStore`. 

Something like this:
```kotlin
private fun setMessageComposeTheme(value: String) {
    skipSaveSettings = true
    generalSettingsManager.setMessageComposeTheme(stringToSubTheme(value))
}

private fun saveSettings() {
    if (skipSaveSettings) {
        skipSaveSettings = false
    } else {
        K9.saveSettingsAsync()
    }
}
```